### DISCUSSION TITLE
Why doesn't UPPAAL try to take all the paths?

### DISCUSSION THREAD

**Discussion by Lqs66** [UPPAALModelChecker/UPPAAL-Meta#276] [Q&A] [answered]
I have added a global variable declaration to my model `int g`.
Then I executed the model in the simulator and I noticed that it always executes along a fixed path.
![image](https://github.com/UPPAALModelChecker/UPPAAL-Meta/assets/62269820/2f17c840-7ada-45d3-b147-0cc89e9a8332)

I found that uninitialized variables are always initialized to 0 by default, which does not match my understanding. Why doesn't the model checker try to traverse all the paths without assigning a random value for g?

**Comment by mikucionisaau** [comment, marked as answer]
Uppaal initializes all variables with `0` unless it has a special range or initializer.
That's because integer variables are treated as part of a discrete state, it does not have an abstract interpretation engine.
Also such treatment would automatically result in much bigger state spaces than most people expect.

If you want a non-deterministic initalization, then you need to model it with non-deterministic edges.
![image](https://github.com/UPPAALModelChecker/UPPAAL-Meta/assets/4868894/4db02e7f-787a-45d2-9cc6-cc1a4f0d9eb8)

**Reply by Lqs66** [reply]
> Uppaal initializes all variables with `0` unless it has a special range or initializer. That's because integer variables are treated as part of a discrete state, it does not have an abstract interpretation engine. Also such treatment would automatically result in much bigger state spaces than most people expect.
> 
> If you want a non-deterministic initalization, then you need to model it with non-deterministic edges. ![image](https://private-user-images.githubusercontent.com/4868894/338846533-4db02e7f-787a-45d2-9cc6-cc1a4f0d9eb8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgyNDY3OTcsIm5iZiI6MTcxODI0NjQ5NywicGF0aCI6Ii80ODY4ODk0LzMzODg0NjUzMy00ZGIwMmU3Zi03ODdhLTQ1ZDItOWNjNi1jYzFhNGYwZDllYjgucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDYxMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA2MTNUMDI0MTM3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OGQ1M2I3MGEzNTYyODMwNzIyYTY4ZjRlODM5NDZjODY0YTg4MzRmOGVlMjg2OWQ4YzI0Y2ExYmNkNDMwNWZhOSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.fnuAfovHD7ibAUdDDq6_2xn1ZDzShMYSsBZ6L71nm2s)

Thank you for your reply.
